### ISSUE TITLE
Allow for `none` SMTP authentication

### ISSUE THREAD

**Issue by D3add3d** [knadh/listmonk#54]
Hello, our SMTP servers are set-up in such way that they do not require any authentication, however putting "plain", username "", password "" or "none" and deleting the username and password completely results in error 535.

Could you please add a way to add SMTP servers that do not require any authentication?

**Comment by knadh** [comment]
https://github.com/knadh/listmonk/commit/047de697707e4a3ea23b53c7a1917f20edc9f8c2 fixes this. The `v0.3.2-alpha` release is out.

Thanks for reporting.

**Comment by D3add3d** [comment]
> 
> 
> [047de69](https://github.com/knadh/listmonk/commit/047de697707e4a3ea23b53c7a1917f20edc9f8c2) fixes this. The `v0.3.2-alpha` release is out.
> 
> Thanks for reporting.

Hello again @knadh ,

your fix allowed me to add the server but I think listmonk is now trying to connect to localhost instead of the SMTP server or it is saying that it is localhost...

My configuration:
```
[app]
address = "0.0.0.0:9000"
root = "https://lists.edu.sh.cvut.cz"
logo_url = ""
favicon_url = ""
from_email = "Vzdelavaci sekce klubu Silicon Hill <edu[AT]sh.cvut.cz>"
notify_emails = ["akadis[AT]sh.cvut.cz"]
concurrency = 100
max_send_errors = 100

...

# Maximum active and idle connections to pool.
max_open = 50
max_idle = 10

# SMTP servers.
[smtp]
    [smtp.sh]
        enabled = true
        host = "smtp.sh.cvut.cz"
        port = "25"

        # cram | plain | empty for no auth
        auth_protocol = ""
        username = ""
        password = ""

        # Maximum time (milliseconds) to wait per e-mail push.
        send_timeout = 5000

        # Maximum concurrent connections to the SMTP server.
        max_conns = 10
...
```

Console output:
```
listmonk@edu:~/listmonk$ ./listmonk
2020/02/05 14:30:35 main.go:113: reading config: config.toml
2020/02/05 14:30:36 main.go:187: loaded SMTP: sh (@smtp.sh.cvut.cz)
⇨ http server started on [::]:9000
2020/02/05 14:30:51 manager.go:166: start processing campaign (test)
2020/02/05 14:30:51 manager.go:349: campaign (test) finished
2020/02/05 14:30:51 manager.go:243: error sending message in campaign test: 504 5.5.2 <localhost>: Helo command rejected: need fully-qualified hostname
2020/02/05 14:30:51 notifications.go:28: error sending admin notification (Finished: test): 504 5.5.2 <localhost>: Helo command rejected: need fully-qualified hostname
2020/02/05 14:30:51 manager.go:243: error sending message in campaign test: 504 5.5.2 <localhost>: Helo command rejected: need fully-qualified hostname
```

**Comment by knadh** [comment]
Looks like your SMTP is configured to only accept an FQDN (eg: mytestsite.com), but the host you're running on has an FQDN set as it's hostname. Is the hostname `localhost`?

The HELO command is sent by Go's smtplib, not listmonk.

**Comment by D3add3d** [comment]
> 
> 
> Looks like your SMTP is configured to only accept an FQDN (eg: mytestsite.com), but the host you're running on has an FQDN set as it's hostname. Is the hostname `localhost`?
> 
> The HELO command is sent by Go's smtplib, not listmonk.

Output of hostname and hostname --fqdn:
```
listmonk@edu:~/listmonk$ hostname
edu
listmonk@edu:~/listmonk$ hostname --fqdn
edu.sh.cvut.cz
```

telnet connection to the SMTP server and `HELO edu.sh.cvut.cz` returns `250 svc2.sh.cvut.cz`

I can't find anything that could be wrong...

**Comment by knadh** [comment]
Figured out the issue. It's to do with the [behaviour of the Go SMTP client](https://github.com/mailhog/MailHog/issues/257). Working on figuring out a fix for this.

```go
package main

import (
	"log"
	"net/smtp"
)

func main() {
	var (
		host = "smtp.sh.cvut.cz:25"
		from = "from@example.com"
		to   = "to@example.com"

		message = `To: "To" <from@example.com>
From: "From" <to@example.com>
Subject: Test

Test message.
`
	)

	if err := smtp.SendMail(host, nil, from, []string{to}, []byte(message)); err != nil {
		log.Fatalf("error sending mail: %v", err)
	}
	log.Println("done")
}
```

```
2020/02/06 11:44:11 error sending mail: 504 5.5.2 <localhost>: Helo command rejected: need fully-qualified hostname
```

**Comment by knadh** [comment]
@D3add3d https://github.com/knadh/listmonk/commit/9a88c2ed7b69bf020ccc95625788017f214a29c3 fixes this (`v0.3.3-alpha`). You'll need to set `hello_hostname` in the config (see sample) to use a custom FQDN.

I've sent a PR to https://github.com/jordan-wright/email/pull/101

**Comment by D3add3d** [comment]
Thank you! That fixed it :) 👍
